src/pubsub.c

```diff
@@ -475,7 +565,29 @@ int pubsubPublishMessageInternal(robj *channel, robj *message, pubsubtype type)
         return receivers;
     }
 
-    /* Send to clients listening to matching channels */
+    /* Send to clients listening to channels matching prefix patterns */
+    raxDescend d;
+    raxDescendStart(&d,server.pubsub_prefixes,channel->ptr,sdslen(channel->ptr));
```

Comment:
Note that the original code passes `channel` to `getDecodedObject()` before accessing its data.

Reply:
I put getDecodedObject in the code that populates pubsub_prefixes so the patterns returned by `raxDescendNext` will already be decoded.

Reply:
Wait that's not exactly it. Your comment are about `channel` itself. I'll decode that too.